test/e2e/fundingManager/BondingCurveTokenRescueE2E.t.sol

```diff
@@ -0,0 +1,257 @@
+                bancorVirtualSupplyBondingCurveFundingManagerMetadata,
+                abi.encode(address(issuanceToken), bc_properties, token)
+            )
+        );
+
+        // Authorizer
+        setUpRoleAuthorizer();
+        moduleConfigurations.push(
+            IOrchestratorFactory_v1.ModuleConfig(
+                roleAuthorizerMetadata, abi.encode(address(this))
+            )
+        );
+
+        // PaymentProcessor
+        setUpSimplePaymentProcessor();
+        moduleConfigurations.push(
+            IOrchestratorFactory_v1.ModuleConfig(
+                simplePaymentProcessorMetadata, bytes("")
+            )
+        );
+    }
+
+    function test_e2e_TokenRescue() public {
+        // address(this) creates a new orchestrator.
+        IOrchestratorFactory_v1.WorkflowConfig memory workflowConfig =
+        IOrchestratorFactory_v1.WorkflowConfig({
+            independentUpdates: false,
+            independentUpdateAdmin: address(0)
+        });
+
+        IOrchestrator_v1 orchestrator =
+            _create_E2E_Orchestrator(workflowConfig, moduleConfigurations);
+
+        FM_BC_Bancor_Redeeming_VirtualSupply_v1 fundingManager =
+        FM_BC_Bancor_Redeeming_VirtualSupply_v1(
+            address(orchestrator.fundingManager())
+        );
+
+        issuanceToken.setMinter(address(fundingManager), true);
+
+        // IMPORTANT
+        // =========
+        // Due to how the underlying rebase mechanism works, it is necessary
+        // to always have some amount of tokens in the fundingmanager.
+        // It's best, if the owner deposits them right after deployment.
+        // uint initialDeposit = 10e18;
+        // token.mint(address(this), initialDeposit);
+        // token.approve(address(fundingManager), initialDeposit);
+        // fundingManager.deposit(initialDeposit);
+
+        // Mint some tokens to alice in order to fund the fundingmanager.
+
+        // Alice will perform a very big initial buy.
+        token.mint(alice, aliceBuyAmount);
+        uint buf_minAmountOut =
+            fundingManager.calculatePurchaseReturn(aliceBuyAmount); // buffer variable to store the minimum amount out on calls to the buy and sell functions
+
+        vm.startPrank(alice);
+        {
+            // Approve tokens to orchestrator.
+            token.approve(address(fundingManager), aliceBuyAmount);
+
+            // Deposit tokens, i.e. fund the fundingmanager.
+            fundingManager.buy(aliceBuyAmount, buf_minAmountOut);
+
+            // After the deposit, alice received some amount of receipt tokens
+            // from the fundingmanager.
+            assertTrue(issuanceToken.balanceOf(alice) > 0);
+        }
+        vm.stopPrank();
+
+        // --------------------------------------------------------------------------------
+        // Start Token Rescue
+
+        // Stop minting of new tokens
+
+        issuanceToken.setMinter(address(fundingManager), false);
```

Comment:
We should also pause the buy/sell operation for the FM here. Of course issuance is not possible anymore but redeeming would still work I believe